### PR TITLE
vote account withdraw authority may change the authorized voter

### LINKED ISSUE #22521: Permit the vote account’s authorized withdrawer to also change the authorized voter
#### Problem
A vote account's authorized voter may only be changed by the existing authorized voter.  The authorized voter is a hot key that has a higher probability of being being compromised than other keys.  If compromised the attacker could prevent a vote account from ever voting again in future, which would likely be an extinction event for any validator that depends on 3rd party stake.

#### Proposed Solution
Permit the vote account’s authorized withdrawer to also change the authorized voter.

The other critical fields in a vote account, commission and validator identity, require the withdraw authority to change.  This is good as that key lives offline in a well setup validator. The withdraw authority is already doing more than just authorizing the withdrawal of SOL from the vote account.

### PR DESCRIPTION
**Pull request by mvines**
Fixes #22521

### CODE REVIEW

File: programs/vote/src/vote_state/mod.rs [hunk 1]
```diff
@@ -921,21 +921,37 @@ pub fn authorize<S: std::hash::BuildHasher>(
...
+
             vote_state.set_new_authorized_voter(
                 authorized,
                 clock.epoch,
                 clock.leader_schedule_epoch + 1,
-                |epoch_authorized_voter| verify_authorized_signer(&epoch_authorized_voter, signers),
+                |epoch_authorized_voter| {
+                    // current authorized withdrawer or authorized voter must say "yay"
+                    if authorized_withdrawer_signer {
+                        Ok(())
```
Comment (mvines) [inline programs/vote/src/vote_state/mod.rs:946]: This is the important part

File: programs/vote/src/vote_state/mod.rs [hunk 2]
```diff
@@ -921,21 +921,37 @@ pub fn authorize<S: std::hash::BuildHasher>(
...
     match vote_authorize {
         VoteAuthorize::Voter => {
+            let authorized_withdrawer_signer = if feature_set
+                .is_active(&feature_set::vote_withdraw_authority_may_change_authorized_voter::id())
+            {
+                verify_authorized_signer(&vote_state.authorized_withdrawer, signers).is_ok()
+            } else {
+                false
+            };
+
```
Comment (mvines) [inline programs/vote/src/vote_state/mod.rs:938]: `vote_state.set_new_authorized_voter()` takes a `&mut self` so can't read `vote_state.authorized_withdrawer` from the "epoch_authorized_voter" closure, hence this setup

**Review by t-nelson — APPROVED** [review]
nice! lgtm

File: programs/vote/src/vote_state/mod.rs [hunk 3]
```diff
@@ -921,21 +921,37 @@ pub fn authorize<S: std::hash::BuildHasher>(
...
             vote_state.set_new_authorized_voter(
                 authorized,
                 clock.epoch,
                 clock.leader_schedule_epoch + 1,
-                |epoch_authorized_voter| verify_authorized_signer(&epoch_authorized_voter, signers),
+                |epoch_authorized_voter| {
+                    // current authorized withdrawer or authorized voter must say "yay"
```
Comment (CriesofCarrots) [inline programs/vote/src/vote_state/mod.rs:944]: nittiest of nits, but it's spelled "yea"

Comment (mvines) [reply programs/vote/src/vote_state/mod.rs:944]: I left it thusly in honour of the late @rob-solana.

cc: https://github.com/solana-labs/solana/blame/master/programs/vote/src/vote_state/mod.rs#L928

Comment (CriesofCarrots) [reply programs/vote/src/vote_state/mod.rs:944]: Aww, hard to argue with that ;)

**Review by joncinque — APPROVED** [review]
looks good!

**Pull request merged**